pkg/apis/autoscaling/register.go

```diff
@@ -83,6 +83,19 @@ const (
 	// scale-to-zero-pod-retention-period global setting.
 	ScaleToZeroPodRetentionPeriodKey = GroupName + "/scaleToZeroPodRetentionPeriod"
 
+	// MetricAggregationAlgorithm is the annotation that can be used for selection
+	// of the algorithm to use for averaging metric data in the Autoscaler.
+	// Since autoscalers are a pluggable concept, this field is only validated
+	// for Revisions that are owned by Knative Pod Autoscaler.
+	// Possible values for KPA are:
+	// - empty/missing or "linear" — linear average over the whole
+	//   metric window (default);
+	// - weightedExponential — weighted average with exponential decay.
+	//   KPA will compute the decay multiplier automatically and it is capped
+	//	 at 0.2 from below. This algorithm might not utilize all the values
+	//   in the window, due to their coefficients being infinitesimal.
+	MetricAggregationAlgorithm = GroupName + "/metricAggregationAlgorithm"
```

Comment:
Should we mark this experimental somehow? Either just in the comment or even in the annotation name/value? I think we want the freedom to muck around with the UX a bit before it's "set" and we wouldn't want anyone insisting on backwards compat before then. If we feel it's actually good-to-go before next release we can always remove the warning. (I guess a counter-argument is this is an annotation anyway but.. I'm not convinced people treat annotations as hints that might go away in this project)

Reply:
I can leave this as a comment.